src/form.ts

```diff
@@ -0,0 +1,173 @@
+const EXCLUDED_TAGS = /\b(?:file|submit|image|reset|button)\b/;
+
+type FormValue = { [ key: string ]: any };
+
+/**
+ * Fills in a DOM form using values from a JavaScript object. Note that fields not specified in the value object will be
+ * cleared.
+ *
+ * @param form the DOM form to set values in
+ * @param object an object containing values for each field in the form
+ *
+ * @example
+ * let formNode = document.getElementById('aForm');
+ * form.fromObject(formNode, { firstName: 'foo', lastName: 'bar' });
+ */
+export function fromObject(form: HTMLFormElement, object: FormValue): void {
+	const elements = form.elements;
+	for (let i = 0; i < elements.length; i++) {
+		const element = <HTMLInputElement> elements[i];
+		const type = element.type;
+		const name = element.name;
+
+		if (!name || EXCLUDED_TAGS.test(type) || element.disabled) {
+			continue;
+		}
+
+		switch (type) {
+			case 'checkbox':
+				element.checked = false;
+				if (name in object) {
+					const checkValue = Array.isArray(object[name]) ? object[name] : [ object[name] ];
+					element.checked = checkValue.indexOf(element.value) !== -1;
+				}
+				break;
+
+			case 'radio':
+				element.checked = (name in object) && object[name] === element.value;
+				break;
+
+			case 'select-multiple':
+				const multiSelectElement: HTMLSelectElement = <any> element;
+				// A multi-select has no selection by default
+				multiSelectElement.selectedIndex = -1;
+				if (name in object) {
+					const selectValue = Array.isArray(object[name]) ? object[name] : [ object[name] ];
+					for (let i = 0; i < multiSelectElement.options.length; i++) {
+						const option = multiSelectElement.options[i];
+						option.selected = selectValue.indexOf(option.value) !== -1;
+					}
+				}
+				break;
+
+			case 'select-one':
+				const selectElement: HTMLSelectElement = <any> element;
+				// A single-select selects the first item by default
+				selectElement.selectedIndex = 0;
+				if (name in object) {
+					const selectValue = object[name];
+					for (let i = 0; i < selectElement.options.length; i++) {
+						const option = selectElement.options[i];
+						option.selected = selectValue.indexOf(selectValue) !== -1;
+					}
+				}
+				break;
+
+			case 'hidden':
+			case 'text':
+			case 'textarea':
+			case 'password':
+				// A text input value is the empty string by default
+				element.value = object[name] || '';
+				break;
+		}
+	}
+}
+
+/**
+ * Gets the value of a form field.
+ */
+function getValue(field: HTMLInputElement): string | string[] {
+	const type = field.type;
+	let value: string | string[];
+
+	if (type === 'radio' || type === 'checkbox') {
+		if (field.checked) {
+			value = field.value;
+		}
+	}
+	else if (field.multiple) {
```

Comment:
This code inherited from Dojo 1 seems pretty weird.  Would `getElementsByTagName` work to simplify?